spec/helpers/location_helper_spec.rb

```diff
@@ -0,0 +1,67 @@
+    end
+
+    it "doesn't truncate shorter names" do
+      name = "Alaska, USA"
+      result = LocationHelper.truncate_name(name)
+      expect(result).to eq("Alaska, USA")
+    end
+  end
+
+  # TODO: These tests would be improved with Factories and SamplesHelper tests
+  describe "#sample_dimensions" do
+    it "gets and formats sample location dimensions" do
+      sample_ids = [1, 2, 3]
+      field_name = "collection_location_v2"
+      mock_filtered = { [nil, "Alaska, USA"] => 1, [nil, "Chania, Chania Municipality, Chania Regional Unit, Region of Crete, Crete, 73135, Greece"] => 1, [nil, "Hanoi, Vietnam"] => 1, [nil, "Redwood City, San Mateo County, California, USA"] => 112, [nil, "Zimbabwe"] => 1 }
+      allow(SamplesHelper).to receive_message_chain(:samples_by_metadata_field, :count).and_return(mock_filtered)
+
+      expected = [{ value: "Alaska, USA", text: "Alaska, USA", count: 1 },
+                  { value: "Chania, Chania Municipality, Chania Regional Unit, Region of Crete, Crete, 73135, Greece", text: "Chania, Chania Municipality, Chania Regional ...", count: 1 },
+                  { value: "Hanoi, Vietnam", text: "Hanoi, Vietnam", count: 1 },
+                  { value: "Redwood City, San Mateo County, California, USA", text: "Redwood City, San Mateo County, California, U...", count: 112 },
+                  { value: "Zimbabwe", text: "Zimbabwe", count: 1 },
+                  { value: "not_set", text: "Unknown", count: 740 }]
+      result = LocationHelper.sample_dimensions(sample_ids, field_name, 856)
+      expect(result).to eq(expected)
+    end
+  end
+
+  # TODO: These tests would be improved with Factories and SamplesHelper tests
+  describe "#project_dimensions" do
+    it "gets and formats project location dimensions" do
+      field_name = "collection_location_v2"
+      mock_filtered = { [nil, "Alaska, USA"] => 1, [nil, "Chania, Chania Municipality, Chania Regional Unit, Region of Crete, Crete, 73135, Greece"] => 1, [nil, "Hanoi, Vietnam"] => 1, [nil, "Redwood City, San Mateo County, California, USA"] => 1, [nil, "Zimbabwe"] => 1 }
+      allow(SamplesHelper).to receive_message_chain(:samples_by_metadata_field, :includes, :distinct, :count).and_return(mock_filtered)
+
+      expected = [{ value: "Alaska, USA", text: "Alaska, USA", count: 1 },
+                  { value: "Chania, Chania Municipality, Chania Regional Unit, Region of Crete, Crete, 73135, Greece", text: "Chania, Chania Municipality, Chania Regional ...", count: 1 },
+                  { value: "Hanoi, Vietnam", text: "Hanoi, Vietnam", count: 1 },
+                  { value: "Redwood City, San Mateo County, California, USA", text: "Redwood City, San Mateo County, California, U...", count: 1 },
+                  { value: "Zimbabwe", text: "Zimbabwe", count: 1 }]
+      result = LocationHelper.project_dimensions([1, 2, 3], field_name)
+      expect(result).to eq(expected)
+    end
+  end
+
+  describe "#filter_by_name" do
```

Comment:
If you really must have this placeholder, add a `pending "<todo description here>"`.
This will generate a warning when running tests.

Reply:
do you think I should just leave it out? or is there something I can still test without pulling in fixtures

```
    samples = samples_with_metadata.includes(metadata: :location)
    # Plain text locations in string_validated_value
    samples.where(metadata: { string_validated_value: query })
           .or(samples.where(metadata: { locations: { name: query } }))
```

Reply:
added the pending 👍